x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportGetTransformStatsAction.java

```diff
@@ -132,28 +142,49 @@ protected void taskOperation(Request request, TransformTask task, ActionListener
 
     @Override
     protected void doExecute(Task task, Request request, ActionListener<Response> finalListener) {
-        final ClusterState state = clusterService.state();
-        TransformNodes.warnIfNoTransformNodes(state);
+        final ClusterState clusterState = clusterService.state();
+        TransformNodes.warnIfNoTransformNodes(clusterState);
+        final DiscoveryNodes nodes = clusterState.nodes();
 
         transformConfigManager.expandTransformIds(
             request.getId(),
             request.getPageParams(),
             request.isAllowNoMatch(),
             ActionListener.wrap(hitsAndIds -> {
-                request.setExpandedIds(hitsAndIds.v2());
-                final TransformNodeAssignments transformNodeAssignments = TransformNodes.transformTaskNodes(hitsAndIds.v2(), state);
+
+                if (hitsAndIds.v2().v2().stream().anyMatch(config -> config.getSource().requiresRemoteCluster())
+                        && (isRemoteClusterClientNode == false)) {
```

Comment:
I think this is not necessary at this level:

if a transform is running, the call is redirected to the node that runs the transform to get _live data_ from that task. If the transform requires remote client, the transform running node must be transform and remote client by definition.

But we hit a problem for `collectStatsForTransformsWithoutTasks`: This is executed on _this_ node.

`transformNodeAssignments` tells you which transforms are `waitingForAssignment` or `stopped`. I would use your added `selectAnyTransformRemoteNode` method to send a request only with non running transforms to an appropriate transform node with or without remote client.